### PR TITLE
Table Reorganization

### PR DESCRIPTION
**Pull request by wavious-wpatty**
Fixes:
  - None

Features:
  - Combined calibration and configuration tables
  - Moved table includes to dedicated directory

### CODE REVIEW

File: include/table/dfi/table.h [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+        uint32_t gb_mode: 4;
+    };
+    uint32_t val;
+} dfi_pipe_cfg_t;
+
+/**
+ * @brief   DFI Clock Enable Pulse Extension
+ *
+ * @details DFI Clock Enable Pulse Extension structure. Used to add cycles to
+ *          end of clock pulse o extend width by configured number of phases.
```
Comment (wavious-jbasista) [inline include/table/dfi/table.h]: typo, add t in to

File: include/table/dfi/table.h [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+        uint32_t ck_clk: 4;
+        uint32_t rd_clk: 4;
+    };
+    uint32_t val;
+} dfi_clken_pext_cfg_t;
+
+/**
+ * @brief   DFI Override Traffic Configuration
+ *
+ * @details DFI Traffic Override Traffic configuration. Configures dynamic clock
```
Comment (wavious-jbasista) [inline include/table/dfi/table.h]: Two traffic in sentence, remove one

File: include/table/dfi/table.h [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+/**
+ * @brief   DFI Pad Enable Pulse Extension
+ *
+ * @details Phase extensions for write / read data and control paths in terms of
+ *          number of phases.
+ *
+ * @note    This structure directly maps to the CSR layout. Please do not
+ *          modify unless the CSR fields change.
+ *
+ * wrd.cs   Write data chip select.
```
Comment (wavious-jbasista) [inline include/table/dfi/table.h:154]: should we add .fields to these?

File: include/table/dfi/table.h [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+        uint32_t val;
+    } rd;
+} dfi_paden_pext_cfg_t;
+
+
+/**
+ * @brief   DFI Frequency Dependent Configuration Structure
+ *
+ * @details Per-Frequency configuration structure for DFI device.
+ */
```
Comment (wavious-jbasista) [inline include/table/dfi/table.h:216]: Missing description of members in struct.

File: include/table/dp/table.h [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+
+/**
+ * @brief Pipeline Bit Configuration Structure
+ *
+ * @details Per frequency configuration data for each bit in the pipeline.
+ *
+ * sdr      sdr dp parameters.
+ * ddr      ddr dp parameters.
+ * qdr      qdr dp parameters.
+
```
Comment (wavious-jbasista) [inline include/table/dp/table.h]: extra line space

File: include/table/dram/table.h [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+ * mr11             Mode Register 11 (CA / DQ ODT).
+ * mr12    Mode Register 12 (CA VREF).
+ * mr14    Mode Register 14 (DQ VREF).
+ * phy_wr_lat       Write Enable Offset (DFI).
+ * phy_wr_en        Write Data Offset (DFI).
+ * phy_rd_en        Read Enable Offset (DFI).
+ * t_sh_train       Setup / Hold cycles for CBT.
+ * ratio            DFI to DRAM Clock ratio.
+ * t_vref_ca_long   CA VREF LONG in cycles.
+ * txsr             Self-Refresh Exit timing.
```
Comment (wavious-jbasista) [inline include/table/dram/table.h]: Move these to the top to match structure.

File: include/table/dram/table.h [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+/**
+ * @brief   DRAM Configuration Structure
+ *
+ * @details Per-Frequency DRAM configuration data.
+ *
+ * mr1              Mode Register 1 (Toggle Preamble).
+ * mr2              Mode Register 2 (Read / Write Latency).
+ * mr11             Mode Register 11 (CA / DQ ODT).
+ * mr12    Mode Register 12 (CA VREF).
+ * mr14    Mode Register 14 (DQ VREF).
```
Comment (wavious-jbasista) [inline include/table/dram/table.h]: alignment

File: include/table/driver/table.h [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+    DRIVER_IMPEDANCE_40
+} driver_impedance_t;
+
+/**
+ * @brief   Driver Common Configuration Structure
+ *
+ * @details Configuration data for Driver common to both slice in TX path.
+ *
+ * code     pointer to calibrated P/N codes. This points to ZQCAL table.
+ * mode     driver mode.
```
Comment (wavious-jbasista) [inline include/table/driver/table.h]: Organize to match structure

File: include/table/pi/tx_table.h [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+ * @brief   TX Phase Interpolator Configuration Frequency Structure
+ *
+ * @details Per frequency configuration data for TX PIs.
+ *
+ * @odr     Configuration for ODR Phase Interpolator.
+ * @qdr     Configuration for QDR Phase Interpolator.
+ * @ddr     Configuration for DDR Phase Interpolator.
+ * @rt      Configuration for RT Phase Interpolator.
+ * @sdr     Configuration for SDR delay match Phase Interpolator.
+ * @dfi     Configuration for DFI delay match Phase Interpolator.
```
Comment (wavious-jbasista) [inline include/table/pi/tx_table.h]: Remove @

File: include/table/receiver/table.h [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+ * C_SIDE_MASK     Receiver C side mask.
+ * BOTH_SIDE_MASK  Max receiver T/C mask.
+ */
+typedef enum receiver_tc_side_mask_t
+{
+    REC_T_SIDE_MASK      = (0x1 << REC_T_SIDE),
+    REC_C_SIDE_MASK      = (0x1 << REC_C_SIDE),
+    REC_BOTH_SIDE_MASK   = (REC_T_SIDE_MASK | REC_C_SIDE_MASK)
+} receiver_tc_side_mask_t;
+/**
```
Comment (wavious-jbasista) [inline include/table/receiver/table.h:39]: add a space above

**Review by wavious-jbasista — CHANGES_REQUESTED** [review]
Just some small comment changes

File: include/table/dfi/table.h (hunk 3 above)
Comment (wavious-wpatty) [reply include/table/dfi/table.h:154]: No, it's okay. It's implied. There's no other .cs, etc.

File: include/table/dfi/table.h (hunk 4 above)
Comment (wavious-wpatty) [reply include/table/dfi/table.h:216]: It's been like that. but I will add.

**Review by wavious-jbasista — APPROVED** [review]
Looks good

**Pull request merged**